### ISSUE TITLE
Introduce `.kraftignore`

### ISSUE THREAD

**Issue by nderjung** [unikraft/kraftkit#135]
This issue tracks ignoring files during either the build process (where files are recognised) or packaging process (where a filesystem is created), a `.kraftignore` file should exclude any glob-patterns from being used.

**Comment by mdsahiloss** [comment]
HI! @nderjung is there anyone working on this issue, If not then I'd like to work on it :)

**Comment by nderjung** [comment]
Hi that's no problem. I just wanted to check in, you were also working on the configurable output format, would you like to swap project?

**Comment by mdsahiloss** [comment]
@nderjung Yes! I'd like to work on this project until `configurable output format` project's decision has been made for the liberary  https://github.com/kubernetes/cli-runtime/tree/master/pkg.

**Comment by mdsahiloss** [comment]
@nderjung shoul I start working on it :)

**Comment by nderjung** [comment]
Hi @MdSahil-oss, this work is dependent on proper volume support within the context of both compiling files (and passing files to the relevant build system); building filesystems (e.g. CPIO); as well as, runtime file system mapping (e.g. 9pfs).

Please hold off on this project until an issue describing proper volume support in KraftKit is put together as it this will implicitly provide greater detail as to how `.kraftignore` will eventually work and how it can be implemented.  I'll have time early next week to put a document together to describe the action items necessary. (:

**Comment by mdsahiloss** [comment]
@nderjung Now, is there possiblity? to implement this feature in Kraftkit ?

**Comment by craciunoiuc** [comment]
Adding a `.kraftignore` might prove confusing to people, as `.dockerignore` already exists

This would help only the `directory` case so not very useful
